src/main/java/org/alfresco/repo/search/impl/lucene/SolrActionNodeReportResult.java

```diff
@@ -0,0 +1,115 @@
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.repo.search.impl.lucene;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * The results of executing a SOLR NODE REPORT action
+ *
+ * @author aborroy
+ * @since 6.2
+ */
+public class SolrActionNodeReportResult extends AbstractJSONAPIResult
+{
+    private static final Log logger = LogFactory.getLog(SolrActionNodeReportResult.class);
+    
+    /**
+     * Parses the JSON to set this Java Object values
+     * @param json JSONObject returned by SOLR API
+     */
+    public SolrActionNodeReportResult(JSONObject json)
+    {
+        try 
+        {
+            processJson(json);
+        }
+        catch (NullPointerException | JSONException e)
+        {
+           logger.info(e.getMessage());
+        }
+    }
+    
+    /* (non-Javadoc)
+     * @see org.alfresco.repo.search.impl.lucene.AbstractSolrActionAPIResult#processCoresInfoJson(org.json.JSONObject)
+     */
+    @Override
+    protected void processCoresInfoJson(JSONObject json) throws JSONException
+    {
+
+        cores = new ArrayList<>();
+        coresInfo = new HashMap<>();
+        
+        if (json.has("report")) 
+        {
+        
+            if (json.has("report")) 
+            {
+            
+                JSONObject coreList = json.getJSONObject("report");
+                JSONArray coreNameList = coreList.names();
+                for(int i = 0; i < coreNameList.length(); i++)
+                {
+                    
+                    String coreName = String.valueOf(coreNameList.get(i));
+                    JSONObject core = coreList.getJSONObject(coreName);
+                    cores.add(coreName);
+                    
+                    Map<String, Object> coreInfo = new HashMap<>();
+                    JSONArray nodesPropertyNameList = core.names();
+                    for (int j = 0; j < nodesPropertyNameList.length(); j++)
+                    {
+                        String propertyName = String.valueOf(nodesPropertyNameList.get(j));
+                        Object propertyValue = core.get(propertyName);
+                        if (propertyValue != JSONObject.NULL)
+                        {
+                            // MBeans Objects are defined as Long types, so we need casting to provide the expected type
+                            if (propertyValue instanceof Integer)
+                            {
+                                propertyValue = Long.valueOf(propertyValue.toString());
+                            }
+                            coreInfo.put(propertyName, propertyValue);
+                        }
+                    }
+                    coresInfo.put(coreName, coreInfo);
```

Comment:
I feel like I've read this code already. Can this be refactored into a helper method?

Reply:
Refactorized!